scoap3/modules/workflows/utils.py

```diff
@@ -34,3 +36,25 @@ def start_compliance_workflow(record):
             data=record.json
         )
     )
+
+
+def delete_halted_workflows_for_doi(doi):
+    """
+    Deletes all workflow that contain the given doi and are in HALTED state.
+
+    The workflow index will only be updated, when a WorkflowObjectModel instance is saved. When a workflow is halted,
+    the connected object's status won't be changed, hence the index won't be updated. Because of all this, we cannot
+    filter for HALTED state in ElasticSearch.
+    """
+
+    current_search_client.indices.refresh("scoap3-workflows-harvesting")
+    search_result = current_search_client.search('scoap3-workflows-harvesting', q='metadata.dois.value:"%s"' % doi)
+
+    workflow_ids = {x['_source']['_workflow']['id_workflow'] for x in search_result['hits']['hits']}
+    for wid in workflow_ids:
+        if wid:
+            w = Workflow.query.get(wid)
```

Comment:
I am not sure this is the right thing. IIRC, the `Workflow` class contains the definition of the workflows itself (all the steps and branches). Same for `WorkflowStatus`. Here I would:
```python
from invenio_workflows import workflow_object_class

w = workflow_object_class.get(wid)
if w.status == ObjectStatus.HALTED:
. . .
```

Reply:
That's basically what I thought, but it is not the case. Not sure if this is an invenio-workflows bug or feature, but when a workflow is halted, the connected objects are in WAITING status, instead of HALTED.
```python3
In [1]: from invenio_workflows import Workflow
In [2]: w =Workflow.query.get('e7dc0b4a-ba86-11e9-830c-02163e01809a')
In [3]: w
Out[3]: <Workflow(name: articles_upload, cre: 2019-08-09 11:20:18.988887, mod: 2019-08-09 11:20:19.964206,id_user: 0, status: WorkflowStatus.HALTED)>
In [4]: w.objects
Out[4]: [<WorkflowObjectModel(id = 264955, id_workflow = e7dc0b4a-ba86-11e9-830c-02163e01809a, status = ObjectStatus.WAITING, id_parent = None, created = 2019-08-09 10:25:39.613958, )]
```